core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -106,7 +115,7 @@ XMLReaderConfig getConfig() {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
-    config.validate();
+    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
     pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_XML_SCHEMA);
     if (!config.containsMacro("tableName") && !Strings.isNullOrEmpty(config.tableName)) {
```

Comment:
this is a relic of the past and will not work well on Hadoop, and will not work at all in a cloud environment.

Please set the widget type for the table name and the table expiry to 'hidden', and remove mention of them from the documentation.

Reply:
Sounds good, fixed